flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/BooleanToStringCastRule.java

```diff
@@ -33,8 +34,10 @@
     private BooleanToStringCastRule() {
         super(
                 CastRulePredicate.builder()
-                        .input(LogicalTypeRoot.BOOLEAN)
-                        .target(LogicalTypeFamily.CHARACTER_STRING)
+                        .predicate(
```

Comment:
how about we introduce a `.target(LogicalType)` instead? It can still translate to this predicate. The input can stay as it is.